cl/lib/pacer_session.py

```diff
@@ -10,6 +11,64 @@
 session_key = "session:pacer:cookies:user.%s"
 
 
+class ProxyPacerSession(PacerSession):
+    """
+    This class overrides the _prepare_login_request and post methods of the
+    PacerSession class to achieve the following:
+
+    - Sets the 'X-WhSentry-TLS' header to 'true' for all requests.
+    - Replaces 'https://' with 'http://' in the URL before making the request.
+    - Uses a proxy server for all requests.
+
+    If the post method is called with a 'headers' argument, it merges the
+    provided headers with the 'X-WhSentry-TLS' header set to 'true'. If no headers
+    argument is provided, it adds a new dictionary with the 'X-WhSentry-TLS' header
+    set to 'true' as the 'headers' argument.
+    """
+
+    def __init__(
+        self, cookies=None, username=None, password=None, client_code=None
+    ):
+        super().__init__(cookies, username, password, client_code)
+        self.proxies = {
+            "http": settings.EGRESS_PROXY_HOST,
+        }
+        self.headers["X-WhSentry-TLS"] = "true"
+
+    def _change_protocol(self, url: str) -> str:
+        """Converts a URL from HTTPS to HTTP protocol.
+
+        By default, HTTP clients create a CONNECT tunnel when a proxy is
+        configured and the target URL uses HTTPS. This doesn't provide the
+        security benefits of initiating TLS from the proxy. To address this,
+        Webhook Sentry provides way of proxying to HTTPS targets. We should:
+
+        1. Change the protocol in the URL to HTTP.
+        2. Set the `X-WhSentry-TLS` header in your request to instruct Webhook
+           Sentry to initiate TLS with the target server.
+
+        https://github.com/juggernaut/webhook-sentry?tab=readme-ov-file#https-target
+
+        Args:
+            url (str): The URL to modify.
+
+        Returns:
+            str: The URL with the protocol changed from HTTPS to HTTP.
+        """
+        return url.replace("https://", "http://")
```

Comment:
This would cause trouble if the URL contained `https://` anywhere outside the protocol. For example:

```
https://foo.com?redirect=https://bar.com
```

That'd get redirected to http://bar.com, and make things less secure. 

Probably the right way to do this is either:
 - use regex and search at the beginning of the string; or
 - use the urlparse library; or
 - get fancy and use Ada

Reply:
Thanks for pointing this out. I've updated the code to use the `urlparse` library